### ISSUE TITLE
Permissions

### ISSUE THREAD

**Issue by stef204** [WangDaYeeeeee/GeometricWeather#58]
Why do you require these permissions:

- read phone status and identity
This is a privacy concern.  Can you please explain in as much detail as possible, why you are requesting this?

- set an alarm
Where in the app does one set an alarm? Why is this required?

**Comment by WangDaYeeeeee** [comment]
- read phone status and identity
Baidu location SDK required this permission, it's mandatory and should be used for statistics and offline location. I did not require these permissions in manifest or java code.

- set an alarm
This permission is used to turn on the alarm when the desktop widget is clicked.

**Comment by stef204** [comment]
Thanks for your feedback.

I believe that the "read phone status and identity" should be either opt-in if one wants the Baidu location service (I certainly don't) and/or turned off by default.

"read phone status and identity" is a privacy risk, it allows, at a minimum:

    
    read your IMEI/IMSI (to identify your device/SIM)
    read your phone number (from the SIM)
    which provider is used (again, the SIM)
    see whether a call is active and if so, phone number of the remote party

I do not think a weather application should require this permission.

Are you able to address this?

**Comment by WangDaYeeeeee** [comment]
You are right. But I have to use Baidu location SDK because of some devices in China don't contain Google Service, I can't use native API to get location information normally. The situation is getting better but still need time.

**Comment by stef204** [comment]
While these issues are worked out, would you consider providing a version (different branch) of your app without the automatic location {Baidu or otherwise)?

One can just add various locations manually as @dimqua just mentioned.

There would also remain issue #53 (non-encrypted connections.)

Addressing both of these issues (#58 and #53), whenever you can, would help to improve the app's security and privacy, 

The app looks great, otherwise.

**Comment by dimqua** [comment]
> some devices in China don't contain Google Service

I suppose most of these devices can't access Google Play Store. It means a version with Baidu SDK should be targeting to Chinese users (e.g. CoolApk users), but not necessarily on Google Play users.

**Comment by papjul** [comment]
Concerns were addressed in this merge request on F-Droid: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/8418
Please let me know if there are still some others.
